CMakeLists.txt

```diff
@@ -4,11 +4,13 @@ set(CMAKE_CXX_STANDARD 20)           # Enable C++20 standard
 
 add_executable(app "main.cpp")
 
+set(DAWN_FETCH_DEPENDENCIES ON)
+add_subdirectory("dawn" EXCLUDE_FROM_ALL)
```

Comment:
This does work, but I would like to direct people toward using the releases instead (https://github.com/google/dawn/releases). Otherwise, people are going to run into all sorts of unnecessary issues with integrating Dawn build files, or not be able to integrate at all because their project doesn't use CMake.

Pretty soon I think there is going to be a flag in Emscripten that lets you enable emdawnwebgpu without having to download it, but it's not there yet. See https://github.com/emscripten-core/emscripten/issues/24260

Otherwise LGTM

Reply:
Per chat discussion we'll keep this building emdawnwebgpu via dawn as a CMake subproject for now. Maybe when we have prebuilt binaries for native platforms on https://github.com/google/dawn/releases we want to switch them all over to make the setup process easier (many fewer chances for setup to go wrong).